relay: Next iteration of refactoring (#2365)
getsentry/sentry-docs: src/docs/product/relay/getting-started.mdx

To create the initial configuration, use Relay's `relay config init` command,
which puts configuration files in the `.relay` folder under the current working
directory.

<Note>

> In Docker, the configuration directory is provided using the standard

Reviewer: I'd leave this active, since we want to tell the user to pass a directory in.

Settings are recorded in `.relay/config.yml`. Note that all configuration values
are optional and default to these settings:

> ```yaml {tabTitle:Run in Docker}

Reviewer: _nit_:

```suggestion
```yaml {tabTitle:Configuration for Docker}
```

Author: Looks like if we do this, tab switching will not be synced across the page :/

```
 INFO  relay::setup > launching relay from config folder .relay
 INFO  relay::setup >   relay mode: managed
 INFO  relay::setup >   relay id: cde0d72e-0c4e-4550-a934-c1867d8a177c

>  INFO  relay::setup >   public key: QPBITKKtKUuEZGGbPke8iufEXAcVrEv6nmWrkRtc3l8

Reviewer: Wanna update the public key to `nDJI79SbEYH9-8NEJAI7ezrgYfoIPW3Bnkg00k1z0fA`, too?

./relay run --config /etc/relay/
```

## Sending a Test Event

Once Relay is running and authenticated with Sentry, send a test event to one of the projects in your organization.

> Get the DSN of your project by navigating to your project settings ( _Settings > Projects > {YOUR_PROJECT_NAME} > Client Keys (DSN)_ ), and selecting one of the existing DSN value, which looks similar to:

Reviewer: ```suggestion
Get the DSN of your project by navigating to your project settings at _Settings > Projects > {YOUR_PROJECT_NAME} > Client Keys (DSN)_, and select one of the existing DSNs, which looks similar to:
```

`https://12345abcdef10111213141516171819@o1.ingest.sentry.io/2345`

Next, replace parts of the DSN to match the address at which Relay is reachable.
For instance, if Relay listens at `http://localhost:3000`, change the protocol
and host of the DSN to:

<code>
<strong><i>http</i></strong>://12345abcdef10111213141516171819@<strong><i>localhost:3000</i></strong>/2345

> </code>

Reviewer: _nit_: Ideally, we could somehow indent this and the previous DSN.

Author: Could you explain?

Reviewer: The current rendering renders this in a paragraph, but it looks just like a line of text that happens to be code. I was wondering if we could somehow indent this paragraph or do something so this sticks out in the reading flow. This is purely visual.

Use the new DSN in your SDK configuration. To test this, you can send a message
with `sentry-cli` ([installation instructions](/cli/installation/)):

```shell
export SENTRY_DSN='http://12345abcdef10111213141516171819@127.0.0.1:3000/2345'
sentry-cli send-event -m 'A test event'
```

In a few seconds, the event should appear in the issues stream in your
project.

## Next Steps

> You have a functioning Relay instance, great! Make sure to check [Operating Guidelines](/product/relay/best-practices/) to learn how to monitor and scale your Relay setup.

Reviewer: While I really like this, the tone of this document is generally very dry and neutral. I'd suggest:

```suggestion
You have a working Relay instance. Make sure to check [Operating Guidelines](/product/relay/best-practices/) to learn how to monitor and scale your Relay setup.
```

> <PageGrid />

Reviewer: This does not render anything anymore.

```suggestion
<PageGrid />
```

./relay run --config /etc/relay/
```

## Sending a Test Event

Once Relay is running and authenticated with Sentry, send a test event to one of the projects in your organization.

Get the DSN of your project by navigating to your project settings at _Settings > Projects > {YOUR_PROJECT_NAME} > Client Keys (DSN)_, and select one of the existing DSNs, which looks similar to:

`https://12345abcdef10111213141516171819@o1.ingest.sentry.io/2345`

Next, replace parts of the DSN to match the address at which Relay is reachable.
For instance, if Relay listens at `http://localhost:3000`, change the protocol
and host of the DSN to:

<p>
  <code>
    <strong><i>http</i></strong>
    ://12345abcdef10111213141516171819@
    <strong><i>localhost:3000</i></strong>
    /2345
  </code>
</p>

Use the new DSN in your SDK configuration. To test this, you can send a message
with `sentry-cli` ([installation instructions](/cli/installation/)):

```shell
export SENTRY_DSN='http://12345abcdef10111213141516171819@127.0.0.1:3000/2345'
sentry-cli send-event -m 'A test event'
```

In a few seconds, the event should appear in the issues stream in your
project.

## Next Steps

> You have a working Relay instance. Make sure to check [Operating Guidelines](/product/relay/operating-guidelines/) to learn how to monitor and scale your Relay setup.

Reviewer 2: ```suggestion
- **[Learn more about our Operating Guidelines](/product/relay/operating-guidelines/)**

With a working Relay instance, you can both monitor and scale your Relay setup.
```
